### PR TITLE
feat(OnyxSelect): Implement truncation for OnyxSelect options

### PR DESCRIPTION
**Pull request by MajaZarkova**
Relates to #733 

Implemented new truncation property using the existing 'TruncationType', which will apply the existing 'onyx-truncation-ellipsis' and 'onyx-truncation-multiline' classes

## Checklist

- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue [hunk 1]
```diff
@@ -388,6 +390,7 @@ const selectInputProps = computed(() => {
 
 .onyx-select-wrapper {
   @include layers.component() {
+    --max-height: 20rem;
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue:393]: the var would probably be overridden directly on <OnyxSelect in the project right?
then maybe enhance the name so it's clear that the max height refers to the flyout part?

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue [hunk 2]
```diff
@@ -360,6 +361,7 @@ const selectInputProps = computed(() => {
                 :active="option.value === activeValue"
                 :icon="option.icon"
                 :density="props.density"
+                :truncation="props.truncation"
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue:364]: could you please add a storybook example? did you test it in combination with the custom-options?

Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue:364]: info: it works properly with custom-options 👍
image

but having one storybook example with multiline flyout would be helpful :)

File: .changeset/chatty-ducks-play.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"sit-onyx": minor
+---
+
+Implement truncation for OnyxSelect options
```
Comment (BoppLi) [inline .changeset/chatty-ducks-play.md:5]: not sure if it _needs_ to be done like that, but I guess the majority of our changelog messages are in the syntax

```suggestion
feat(OnyxSelect): Implement truncation for OnyxSelectOptions
```

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx [hunk 4]
```diff
@@ -93,6 +98,29 @@ test.describe("Empty screenshots", () => {
   });
 });
 
+test.describe("Truncated options screenshots", () => {
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx:101]: could you hover on one of the options so we can verify that the hover background covers the whole multiline option?

**Review by BoppLi — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx [hunk 5]
```diff
@@ -93,6 +98,29 @@ test.describe("Empty screenshots", () => {
...
+    component: (column, row) => (
+      <div>
+        <OnyxSelect
+          label="Label"
+          listLabel="List label"
+          options={MOCK_LONG_LABELED_OPTIONS}
+          density={column}
+          truncation={row}
+        />
+      </div>
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx:116]: the div should be redundant

```suggestion
      <OnyxSelect
        label="Label"
        listLabel="List label"
        options={MOCK_LONG_LABELED_OPTIONS}
        density={column}
        truncation={row}
      />
```

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue (hunk 2 above)
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue:364]: shouldn't multiline also work with the same truncation?
image

File: packages/sit-onyx/src/components/OnyxSelect/types.ts [hunk 6]
```diff
@@ -84,6 +89,10 @@ export type OnyxSelectProps<TValue extends SelectOptionValue = SelectOptionValue
      * If you want to use a button instead, use the `optionsEnd` slot.
      */
     lazyLoading?: SelectLazyLoading;
+    /**
+     * How to truncate the option label / text content if it exceeds the max width.
+     */
+    truncation?: TruncationType;
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/types.ts:95]: question: shouldn't we have the same pattern as in the checkboxGroup and let the truncation be declared on the options-level? BaseSelectOption is shared between checkboxGroup and select and already offers `truncation` 
```
export type SelectOption<TValue extends SelectOptionValue = SelectOptionValue> = Pick<
  BaseSelectOption<TValue>,
  "value" | "label" | "disabled" | "truncation"
> 
```

File: packages/sit-onyx/src/components/OnyxSelectOption/types.ts [hunk 7]
```diff
@@ -14,4 +15,8 @@ export type OnyxSelectOptionProps = Pick<OnyxListItemProps, "active" | "density"
    * Optional icon.
    */
   icon?: string;
+  /**
+   * How to truncate the option label / text content if it exceeds the max width.
+   */
+  truncation?: TruncationType;
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelectOption/types.ts:21]: see other comment, OnyxCheckbox e.g. inherits `truncation` from `BaseSelectOption`

**Review by BoppLi — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxSelect/types.ts (hunk 6 above)
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxSelect/types.ts:95]: In this case, what happens with the checkAll truncation?

Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxSelect/types.ts:95]: I think the check all will always be ellipsis but it's not expected to have such a long text that truncation gets relevant

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx [hunk 8]
```diff
@@ -34,6 +34,12 @@ const MOCK_LONG_LABELED_OPTIONS = Array.from({ length: 10 }, (_, index) => ({
   label: `Long labeled option ${index + 1} `.repeat(4),
 })) satisfies SelectOption[];
 
+const MOCK_MULTILINE_LONG_LABELED_OPTIONS = Array.from({ length: 10 }, (_, index) => ({
+  value: index,
+  truncation: "multiline",
+  label: `Long labeled option ${index + 1} `.repeat(4),
+})) satisfies SelectOption[];
+
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx:42]:
```suggestion
MOCK_LONG_LABELED_OPTIONS.map(option => ({ ...option, truncation: "multiline" }) satisfies SelectOption[];
```

**Review by BoppLi — APPROVED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png
Comment (BoppLi) [inline packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png:1]: the compact ellipsis is missing the highlighting :O

Comment (BoppLi) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png:1]: maybe try to only select by part of the text, like only "Long labeled option 1", you don't need to have the full string as long as there is a unique match

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx [hunk 9]
```diff
@@ -93,6 +103,30 @@ test.describe("Empty screenshots", () => {
...
+        listLabel="List label"
+        options={
+          row === "ellipsis" ? MOCK_LONG_LABELED_OPTIONS : MOCK_MULTILINE_LONG_LABELED_OPTIONS
+        }
+        density={column}
+      />
+    ),
+    beforeScreenshot: async (component) => {
+      await openFlyout(component);
+      const option = component.getByText(`Long labeled option 1 `.repeat(4));
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.ct.tsx:124]: I'm referring to here :)

```suggestion
      const option = component.getByText(`Long labeled option 1 `);
```

**Review by BoppLi — DISMISSED** [review]
everything except the one screenie is perfect now ^^

File: packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png
Comment (BoppLi) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png:1]: see comment below

Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png:1]: I updated the screenshots and it's fixed now :)

**Review by BoppLi — APPROVED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png
Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-truncated--firefox-linux.png:1]: I was missing 'await' in from of .hover()

**Review by BoppLi — APPROVED** [review]

**Pull request merged**